### ISSUE TITLE
mxnet::cpp::NDArray::WaitAll() take about 160ms

### ISSUE THREAD

**Issue by aaron900813** [apache/incubator-mxnet#13245]
here are codes：
    tValRestart;
    net271_executor->Forward(false);
    std::cout << "Forward use " << tValDuration << " ms" << std::endl; 
    tValRestart;
    auto targetx = net271_executor->outputs[0].Copy(global_cpu_ctx);
    auto targety = net271_executor->outputs[1].Copy(global_cpu_ctx);
    auto targetw = net271_executor->outputs[2].Copy(global_cpu_ctx);
    auto targeth = net271_executor->outputs[3].Copy(global_cpu_ctx);
    auto softmax_score = net271_executor->outputs[4].Copy(global_cpu_ctx);
    auto penalty_t = net271_executor->outputs[5].Copy(global_cpu_ctx);
    std::cout << "copy cpu use " << tValDuration << " ms" << std::endl; 
    tValRestart;
    mxnet::cpp::NDArray::WaitAll();
    std::cout << "waitall use " << tValDuration << " ms" << std::endl; 

    Is that normal? 6 outputs are float-type data

**Comment by zachgk** [comment]
@mxnet-label-bot add [C++, Performance, Pending Requester Info]

Hi @aaron900813, can you help provide some info about your environment?
## Environment info (Required)

```
What to do:
1. Download the diagnosis script from https://raw.githubusercontent.com/apache/incubator-mxnet/master/tools/diagnose.py
2. Run the script using `python diagnose.py` and paste its output here.

```

## Build info (Required if built from source)

Compiler (gcc/clang/mingw/visual studio):

MXNet commit hash:
(Paste the output of `git rev-parse HEAD` here.)

Build config:
(Paste the content of config.mk, or the build command.)

**Comment by chinakook** [comment]
Yes, building with CMake and building with Makefile are different.

**Comment by zachgk** [comment]
@mxnet-label-bot remove [Pending Requester Info]
